dot/parachain/types/statement.go

```diff
@@ -69,3 +72,76 @@ type UncheckedSignedFullStatement struct {
 	// The signature by the validator of the signed payload.
 	Signature ValidatorSignature `scale:"3"`
 }
+
+// SigningContext is a type returned by runtime with current session index and a parent hash.
+type SigningContext struct {
+	/// current session index.
+	sessionIndex SessionIndex //nolint:unused
+	/// hash of the parent.
+	parentHash common.Hash //nolint:unused
+}
+
+// Sign signs the Statement using the provided keystore, signing context, and validator key,
+// returning the unchecked signed full statement and any errors.
+func (u UncheckedSignedFullStatement) Sign(
+	keystore keystore.Keystore,
+	signingContext SigningContext,
+	key ValidatorID,
+) (UncheckedSignedFullStatement, error) {
```

Comment:
I wanted to return a new (non-pointer) instance with a sign in it,
If you want me to use a pointer, we can use a pointer receiver, add a sign and return only an error in case of failure.

Reply:
removed this method.